core/src/main/java/org/opensearch/sql/executor/execution/QueryPlanFactory.java

```diff
@@ -62,39 +65,62 @@ public void onFailure(Exception e) {
   /**
    * Create QueryExecution from Statement.
    */
-  public AbstractPlan create(
+  public AbstractPlan createContinuePaginatedPlan(
       Statement statement,
       Optional<ResponseListener<ExecutionEngine.QueryResponse>> queryListener,
       Optional<ResponseListener<ExecutionEngine.ExplainResponse>> explainListener) {
     return statement.accept(this, Pair.of(queryListener, explainListener));
   }
 
+  /**
+   * Creates a ContinuePaginatedPlan from a cursor.
+   */
+  public AbstractPlan createContinuePaginatedPlan(String cursor, boolean isExplain,
```

Comment:
1. two createContinuePaginatedPlan function defined, what is the difference?
2. can cursor be an CursorStatement? then we can resuse the create interface.

Reply:
1. One is used when you send a paging query (query with fetch_size) and another one is used on cursor requests. Please, see code snippet below: https://github.com/opensearch-project/sql/blob/ce509b0808fdd906c0c29f008566fddeb722a3f7/sql/src/main/java/org/opensearch/sql/sql/SQLService.java#L64-L87
2. Probably yes. Is it a blocker for now or it can be done later?